includes/Admin/Admin_AJAX.php

```diff
@@ -248,4 +279,107 @@ protected function has_runtime_check( array $checks ) {
 
 		return false;
 	}
+
+	/**
+	 * Format results for the AJAX response.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $results The Check Results containing errors and warnings.
+	 *
+	 * @return array An array of results.
+	 */
+	protected function format_results( Check_Result $results ) {
+		// Get errors and warnings from the results.
+		$errors      = $results->get_errors();
+		$warnings    = $results->get_warnings();
+		$all_results = array();
+
+		// Go over all files with errors first and print them, combined with any warnings in the same file.
+		foreach ( $errors as $file_name => $file_errors ) {
+			$file_warnings = array();
+
+			if ( isset( $warnings[ $file_name ] ) ) {
+				$file_warnings = $warnings[ $file_name ];
+				unset( $warnings[ $file_name ] );
+			}
+
+			$file_results = $this->flatten_file_results( $file_name, $file_errors, $file_warnings );
+			$all_results  = array_merge( $all_results, $file_results );
+		}
+
+		// If there are any files left with only warnings, print those next.
+		foreach ( $warnings as $file_name => $file_warnings ) {
+			$file_results = $this->flatten_file_results( $file_name, array(), $file_warnings );
```

Comment:
Why do we need to "flatten" the results? I don't see why that's needed. If it is helpful for the output, we should probably do that in the other issue in JS, but I'm not sure why we would do that.

I'm not even sure we need to do anything to the formatting here. Shouldn't this be the responsibility of the JS? The way this data is best formatted depends on the way it is being rendered, and that should be decoupled from the AJAX endpoint. I think we could probably avoid all this logic and instead simply return `errors` and `warnings` with the raw data in the AJAX response.

Reply:
@felixarntz this was part of the AC's to format the results in a particular way. However, I agree that we should decouple the logic here and return the raw `errors` and `warnings`. The formatting methods have been removed.